File: Backend/connecthub/src/main/java/com/example/connecthub/service/UserService.java
package com.example.connecthub.service;

import java.util.List;

import com.example.connecthub.dto.request.RegisterRequest;
import com.example.connecthub.dto.request.UpdateProfileRequest;
import com.example.connecthub.dto.response.UserProfileResponse;
import com.example.connecthub.dto.response.UserResponse;
import com.example.connecthub.dto.response.UserSearchResponse;

public interface UserService {

    UserResponse register(RegisterRequest request);

    UserProfileResponse getCurrentUserProfile(String email);

    UserProfileResponse updateProfile(
            String email,
            UpdateProfileRequest request);

    List<UserSearchResponse> searchUsers(
            String name,
            String college,
            String profession,
            String location,
            Long interestId);
}

File: frontend/src/services/userService.js
import client from '../api/client';

export const userService = {
  searchUsers: async (filters = {}) => {
    const params = {};
    if (filters.name?.trim()) params.name = filters.name.trim();
    if (filters.college?.trim()) params.college = filters.college.trim();
    if (filters.profession?.trim()) params.profession = filters.profession.trim();
    if (filters.location?.trim()) params.location = filters.location.trim();
    if (filters.interestId) params.interestId = filters.interestId;

    const response = await client.get('/users/search', { params });
    return response.data; // List<UserSearchResponse>
  },
};
